lib/galaxy/tools/__init__.py

```diff
@@ -2773,7 +2773,7 @@ def check_dataset_instance(input_dataset):
 
             if self.require_dataset_ok:
                 if input_dataset.state != input_dataset.dataset.states.OK:
-                    raise ValueError("Tool requires inputs to be in valid state.")
+                    raise ValueError("Tool requires inputs to be in valid state, but {} is in state '{}".format(input_dataset, input_dataset.state))
```

Comment:
Should this be
```suggestion
                    raise ValueError("Tool requires inputs to be in valid state, but dataset {} is in state '{}'".format(input_dataset.id, input_dataset.state))
```
?

Reply:
I was hoping to get the __repr__ of the HDA

Reply:
(because this may not be an HDA)

Reply:
OK, I've updated the suggestion.